src/render/root.js

```diff
@@ -3,17 +3,57 @@ const lead = require('./lead')
 const logic = require('./logic')
 const options = require('./options')
 
-/**
- * @param {*|Array<*>} result
+/*
+ * @param {*|Array<*>} _results
  * @return {string}
  */
-function root(result) {
-  const results = Array.isArray(result) ? result : [result]
+function root(_results) {
+  const results = Array.isArray(_results) ? _results : [_results]
+  const isMultipleResults = results.length > 1
   const [main] = results
-  const logicMethods = results.map((result) => logic(result))
 
-  return [lead(main), imports(main.imports), options(main), ...logicMethods]
-    .filter((item) => item)
+  const resolvedImports = {}
+  const logicFunctions = []
+  const functionNames = []
+  let defaultExported = false
+
+  for (const result of results) {
+    let nameIndex = ''
+    const defaultExport = !defaultExported && result.defaultExport
+    if (defaultExport) {
+      defaultExported = true
+    }
+    let { exportAs: functionName } = result
+    while (functionNames.includes(`${functionName}${nameIndex}`)) {
+      nameIndex = nameIndex === '' ? 0 : nameIndex + 1
+    }
+    functionName += nameIndex
+    functionNames.push(functionName)
+
+    logicFunctions.push(
+      [
+        isMultipleResults ? lead(result) : null,
+        logic({ ...result, exportAs: functionName }, defaultExport),
+      ]
+        .filter(item => item)
+        .join('\n\n')
+    )
+
+    // build imports object
+    Object.entries(result.imports || {}).forEach(([key, value]) => {
+      if (value === true || resolvedImports[key] === undefined) {
+        resolvedImports[key] = value
+      }
+    })
+  }
+
+  return [
+    !isMultipleResults ? lead(main) : null,
+    imports(resolvedImports),
+    options(main),
+    ...logicFunctions,
+  ]
+    .filter(item => item)
     .join(`\n\n`)
 }
```

Comment:
Come to think of it maybe it would make sense to not only validate the archives individually but also the set. If I'm not mistaken the current implementation would not complain if there were multiple archives configured to export default right?

Perhaps we should add a simple validation rule that checks the set then also runs each individual archive like today

```js
function convert() {
  return normalize(archives, options)
    |> validate // validates both the set and also individual archives
    |> parse 
    |> resolveImports
    |> render
}
```

Reply:
The general idea is to allow an advanced user to convert several HAR objects into one script (rather than have to convert them one by one and then stitch them together). The intended output of a "multi conversion" is multiple test scenarios, excluding options. 

> Rather than having this logic in render I think it makes more sense to have it all the way up in convert.js making the flow

I'm not sure I understand

> Come to think of it maybe it would make sense to not only validate the archives individually but also the set.

I don't see a case where two archives could pass validation as a single but cause a fail as a set (unless we're building scenario options). 

> If I'm not mistaken the current implementation would not complain if there were multiple archives configured to export default right?

There can only be one default export per script (no matter the configuration). 

> Perhaps we should add a simple validation rule that checks the set then also runs each individual archive like today